public/app/features/trails/MetricScene.tsx

```diff
@@ -129,11 +158,13 @@ export class MetricActionBar extends SceneObjectBase<MetricActionBarState> {
       <Box paddingY={1}>
         <div className={styles.actions}>
           <Stack gap={2}>
-            <ToolbarButton variant={'canvas'} icon="compass" tooltip="Open in explore (todo)" disabled>
-              Explore
-            </ToolbarButton>
-            <ToolbarButton variant={'canvas'}>Add to dashboard</ToolbarButton>
-            <ToolbarButton variant={'canvas'} icon="share-alt" tooltip="Copy url (todo)" disabled />
+            <ToolbarButton
+              variant={'canvas'}
+              icon="compass"
+              tooltip="Open in explore"
+              onClick={onExplore}
+            ></ToolbarButton>
+            <ShareTrailButton trailUrl={getUrlForTrail(trail)} />
```

Comment:
would just pass in the trail here and call getUrlForTrail inside onShare (so it's lazy)